// mup/lib/server-info.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.parseOutput = parseOutput;
exports.default = serverInfo;

var _utils = require('./utils');

var prefix = '<============mup-var-start========';
var suffix = '================mup-var-stop=====>';
var codeSeperator = 'mup-var-code=======';
function generateVarCommand(name, command) {
  return '\n  echo "' + prefix + name + suffix + '"\n  ' + command + ' 2>&1  \n  echo "' + codeSeperator + '"\n  echo $?\n  ';
}

function parseOutput(output) {
  console.log(output);
  var name = output.split(prefix)[1].split(suffix)[0];
  var commandOutput = output.split(suffix)[1].split(codeSeperator)[0];
  return {
    name: name.trim(),
    output: commandOutput.trim(),
    code: output.split(codeSeperator)[1].trim()
  };
}

function serverInfo(vars, servers) {
  var server = servers[0];
  var command = generateVarCommand('test', 'ls /opt');
  (0, _utils.runSSHCommand)(server, command).then(function (result) {
    console.log(result);
    console.log(parseOutput(result.output));
  }).catch(console.log);
}
